src/fsharp/fsi/fsi.fs

```diff
@@ -2358,6 +2358,9 @@ type FsiEvaluationSession (fsiConfig: FsiEvaluationSessionHostConfig, argv:strin
     let defaultFSharpBinariesDir = System.AppDomain.CurrentDomain.BaseDirectory
     let currentDirectory = Directory.GetCurrentDirectory()
 #endif
+    // When used as part of FCS we cannot assume the current process is fsi.exe
+    // So we try to fallback to the default compiler dir.
+    let defaultFSharpBinariesDir = Internal.Utilities.FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(None).Value
```

Comment:
Hi @matthid 

I'm concerned that this is a breaking change (since it overrides the value on line 2358), and also doesn't really match the common uses of FSharp.Compiler.Service, where the DLL is often used in a standalone way with a fixed referenced FSharp.Core.dll in the same directory.

Could you instead add an optional parameter "useSystemBinaries" to FsiEvaluationSession which, if set to true, uses this value for defaultFSharpBinariesDir